src/test/java/mate/academy/bookshop/controller/OrderControllerTest.java

```diff
@@ -0,0 +1,506 @@
+        OrderItemResponseDto actual = objectMapper.readValue(result
+                        .getResponse().getContentAsString(),
+                OrderItemResponseDto.class);
+
+        assertNotNull(actual);
+        assertEquals(expected, actual);
+    }
+
+    private void setAuthentication(User user) {
+        UserDetails userDetails = withUsername(user.getEmail())
+                .password(user.getPassword())
+                .authorities(user.getRoles().stream()
+                        .map(role -> new SimpleGrantedAuthority(role.getName().name()))
+                        .toArray(SimpleGrantedAuthority[]::new))
+                .build();
+
+        UsernamePasswordAuthenticationToken authToken = new UsernamePasswordAuthenticationToken(
+                userDetails, null, userDetails.getAuthorities());
+
+        SecurityContextHolder.getContext().setAuthentication(authToken);
+    }
+
+    private void compareOrderItems(Set<OrderItemResponseDto> expected,
+                                   Set<OrderItemResponseDto> actual) {
+        assertTrue(expected.stream()
+                .allMatch(expectedItem -> actual.stream()
+                        .anyMatch(actualItem ->
+                                EqualsBuilder.reflectionEquals(expectedItem,
+                                        actualItem,
+                                        "id"))));
+    }
+
+    private OrderResponseDto createOrderResponseDto() {
+
+        OrderItemResponseDto itemResponseDto1 = new OrderItemResponseDto(
+                1L,
+                3L,
+                BigDecimal.valueOf(12.99),
+                2
+        );
+
+        OrderItemResponseDto itemResponseDto2 = new OrderItemResponseDto(
+                2L,
+                2L,
+                BigDecimal.valueOf(11.99),
+                1
+        );
+
+        OrderItemResponseDto itemResponseDto3 = new OrderItemResponseDto(
+                3L,
+                1L,
+                BigDecimal.valueOf(10.99),
+                3
+        );
+
+        OrderResponseDto responseDto = new OrderResponseDto(
+                1L,
+                user.getId(),
+                Set.of(itemResponseDto1, itemResponseDto2, itemResponseDto3),
+                LocalDateTime.now(),
+                BigDecimal.valueOf(12.99 * 2 + 10.99 * 3 + 11.99),
+                Status.PENDING.name()
+        );
+        return responseDto;
+    }
+
+    private void compareExpectedListToActualList(List<OrderResponseDto> expected,
+                                                 List<OrderResponseDto> actual) {
+        for (int i = 0; i < expected.size(); i++) {
+            OrderResponseDto expectedOrder = expected.get(i);
+            OrderResponseDto actualOrder = actual.get(i);
+
+            assertTrue(EqualsBuilder.reflectionEquals(expectedOrder, actualOrder,
+                    "orderDate", "orderItems"));
+            compareOrderItems(expectedOrder.orderItems(), actualOrder.orderItems());
+        }
+    }
```

Comment:
there are methods that provide you with the ability to assert collections in different ways, ignoring order, or comparing only the fields that you want, etc. check the `assertThat(collection).ignoringFields(...).ignoringCollectionsOrder().isEqualTo(anotherCollection)` or something similar, I am sure you'll be able to find something that fits your needs, so you do not have to write this custom assertion methods